insightconnect_plugin_runtime/helper.py

```diff
@@ -302,8 +303,7 @@ def exec_command(command):
     """Return dict with keys stdout, stderr, and return code of executed subprocess command."""
     try:
         p = subprocess.Popen(
-            command,
-            shell=True,
+            shlex.split(command),
```

Comment:
Do we need to join this back together? I believe split only breaks the string into a list, but does not do any escaping, whereas join does: 

https://docs.python.org/3/library/shlex.html#shlex.join
`The returned value is shell-escaped to protect against injection vulnerabilities`

Reply:
Or even only quote() (so we don't need to parse the string first) https://docs.python.org/3/library/shlex.html#shlex.quote
```suggestion
            shlex.quote(command),
```

Reply:
I tried running dig plugin with `shlex.join(shlex.split(command))` and `shlex.quote(command)`  and getting `No such file or directory` error. I think this is due to `/usr/bin/dig google.com ANY` whole being considered as a single command. I mean `google.com` and `ANY` are not being considered as parameters to `/usr/bin/dig`. When we only use `shlex.split(command)` it split the command into list [/usr/bin/sh', 'google.com', 'ANY'] and consider 0th index as the command and rest elements of list as parameters. 

I tried using `shlex.join(shlex.split(command))` and  `shlex.quote(command)` as below and getting permission denied error. I am not sure why we are getting here permission denied error. `shlex.split(command) is working fine.

Could you please give me some sample input which you think can cause issue with shlex.split()? 
```
>>> import shlex
>>> import subprocess
>>> filename = 'export.icon'
>>> command = f'ls -l {filename}'
>>> remote_command = shlex.quote(command)
>>> p = subprocess.Popen(remote_command,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True,)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: "'ls -l export.icon'"
>>>
>>>
>>>
>>>
>>> import shlex
>>> import subprocess
>>> filename = 'export.icon'
>>> command = f'ls -l {filename}'
>>> remote_command = shlex.join(shlex.split(command))
>>> p = subprocess.Popen(remote_command,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True,)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: 'ls -l export.icon'
>>>
>>>
>>> import shlex
>>> import subprocess
>>> filename = 'export.icon'
>>> command = f'ls -l {filename}'
>>> remote_command = shlex.split(command)
>>> p = subprocess.Popen(remote_command,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True,)
>>> p.stdout.read()
b'-rwxrwxrwx  1 gusingh  TOR\\Domain Users  5396 Jul 20 18:21 export.icon\n'
>>>
```